apis/v1alpha2/clusternetworkpolicy_types.go

```diff
@@ -316,29 +315,49 @@ type ClusterNetworkPolicyIngressPeer struct {
 	Pods *NamespacedPod `json:"pods,omitempty"`
 }
 
-// ClusterNetworkPolicyPort describes how to select destination network ports.
-// Exactly one field must be set.
+// ClusterNetworkPolicyProtocol describes how to select traffic by
+// protocol-specific attributes.
+//
+// +kubebuilder:validation:XValidation:rule="!(self.protocol in ['TCP', 'UDP', 'SCTP']) || self.port != nil",message="port must be specified for protocols that support ports"
+type ClusterNetworkPolicyProtocol struct {
+	// Protocol is the network protocol (TCP, UDP, or SCTP) which
+	// traffic must match. If not specified, this field defaults
+	// to TCP.
+	//
+	// +kubebuilder:default=TCP
+	Protocol corev1.Protocol `json:"protocol,omitempty"`
+
+	// Specific port to match against.
+	//
+	// +optional
+	Port *ClusterNetworkPolicyPort `json:"port,omitempty"`
```

Comment:
is Protocol defaulted if Port is nil? I'm confused on how this will work since Port is optional